lib/backend/watchersyncer/watchersyncer_test.go

```diff
@@ -98,6 +98,51 @@ var _ = Describe("Test the backend datastore multi-watch syncer", func() {
 		rs.ExpectStatusUpdate(api.InSync)
 	})
 
+	It("should not change status if watch returns multiple ErrorOperationNotSupported errors", func() {
+		rs := newWatcherSyncerTester([]watchersyncer.ResourceType{r1})
+		rs.ExpectStatusUpdate(api.WaitForDatastore)
+		rs.clientListResponse(r1, emptyList)
+		rs.ExpectStatusUpdate(api.ResyncInProgress)
+		rs.ExpectStatusUpdate(api.InSync)
+		rs.clientWatchResponse(r1, notSupported)
+		rs.ExpectStatusUnchanged()
+		rs.clientWatchResponse(r1, notSupported)
+		rs.ExpectStatusUnchanged()
+		rs.clientWatchResponse(r1, notSupported)
+		rs.ExpectStatusUnchanged()
+		rs.clientWatchResponse(r1, notSupported)
+		rs.ExpectStatusUnchanged()
+		rs.clientWatchResponse(r1, notSupported)
+		rs.ExpectStatusUnchanged()
+	})
+
+	It("should return WaitForDatastore on multiple consecutive watch errors", func() {
+		rs := newWatcherSyncerTester([]watchersyncer.ResourceType{r1})
+		rs.ExpectStatusUpdate(api.WaitForDatastore)
+		rs.clientListResponse(r1, emptyList)
+		rs.ExpectStatusUpdate(api.ResyncInProgress)
+		rs.ExpectStatusUpdate(api.InSync)
+		rs.clientWatchResponse(r1, genError)
+		rs.ExpectStatusUnchanged()
+		rs.clientListResponse(r1, genError)
+		rs.ExpectStatusUnchanged()
+		rs.clientListResponse(r1, genError)
+		rs.ExpectStatusUnchanged()
+		rs.clientListResponse(r1, genError)
+		rs.ExpectStatusUpdate(api.WaitForDatastore)
+
+		// Now get back in-sync and assert that multiple errors up to the threshold does not update status.
+		rs.clientListResponse(r1, emptyList)
+		rs.ExpectStatusUpdate(api.ResyncInProgress)
+		rs.ExpectStatusUpdate(api.InSync)
+		rs.clientWatchResponse(r1, genError)
```

Comment:
Might be nice to encode this into a loop based off `ErrorThreshold` so that we don't need to modify this test if we modify that threshold.